pkg/specgen/generate/container.go

```diff
@@ -272,6 +272,28 @@ func CompleteSpec(ctx context.Context, r *libpod.Runtime, s *specgen.SpecGenerat
 	return warnings, nil
 }
 
+// FinishDeviceWeights takes the temporary representation of the device
```

Comment:
Why not include this in `parseWeightDevices`?

Reply:
I just imitated `FinishThrottleDevices()` without thought.

As I see the history now, `FinishThrottleDevices()` is called in `MakeContainer()` because throttle devices can be configured from an infra container configuration when a pod in created with `--device-read-bps` (#11686).
For a case similar where `--blkio-weight-device` is introduced for a pod in the future, it might be better to implement like this.

Reply:
While throttle values are absolute bps, device weights are relative. The weights cannot be set from an infra container like throttle values. So, it would be better to set device weights at parsing as suggested.

Reply:
Sorry for changing again and again. Finally I got this function called in `CompleteSpec()` because `parseWeightDevices` can be used in a remote client.

Reply:
Basically, I would prefer not to have something modify the SpecGenerator after it has been passed to the server; ideally we'd do it as part of creating the SpecGenerator, but it sounds like that would break the remote client. Instead, can we do this as part of `MakeContainer` and directly modify the OCI spec to be returned, not the SpecGenerator itself?